Lunchtime Buffet

TimothyRogers opened the pull request:
This branch will be version 1.x based on significant changes. 

**Not Ready To Merge**

MDCHAMP committed:
cull the zoo

MDCHAMP committed:
clean up the run

TimothyRogers committed:
Housekeeping

Sort out test discovery failing
Add greedy selection

TimothyRogers committed:
Working through unit tests

TimothyRogers committed:
Initialisers work with unit tests

TimothyRogers committed:
Test Coverage

tech.py - fully tested
PSO/QPSO - tested

TimothyRogers committed:
ABC Crossover function

TimothyRogers committed:
ABC Passing Tests

MDCHAMP committed:
Some gardening and TODOs

MDCHAMP committed:
DE, SADE, PSO, QPSO

MDCHAMP committed:
better PSO hypers

MDCHAMP committed:
Random Search

MDCHAMP committed:
more tests and cov

MDCHAMP committed:
more tests (not all pass)

MDCHAMP committed:
pre_step is not a class method

MDCHAMP committed:
always track the global best

MDCHAMP committed:
PAO in the buffet

MDCHAMP committed:
test expm and obj_score sanitisation

MDCHAMP committed:
docs.

MDCHAMP committed:
qol changes to API

MDCHAMP committed:
minor docs changes

MDCHAMP committed:
docs for benchmarks

MDCHAMP committed:
minor docs changes

MDCHAMP committed:
fix test seed in 1 place only

MDCHAMP committed:
docs for optimisers

MDCHAMP committed:
rm ABC (for now...)

MDCHAMP committed:
Gardening

MDCHAMP committed:
more gardening

MDCHAMP committed:
All first draft docs in

MDCHAMP commented:
Have finally gotten around to adding docs.... probably there is a lot of scope to impove but at least it is something to get us going. Will test a local bulid this week and then can think about render engines / hosting options.

TimothyRogers committed:
Add GWO

Grey wolf optimiser added.
Slackened bounds on test_can_optimise:
PSO failing (it being useless, not a bug)

MDCHAMP committed:
version bump in setup
